src/dev/i18n_tools/bin/run_i18n_integrate.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import chalk from 'chalk';
+import { Listr } from 'listr2';
+
+import { createFailError } from '@kbn/dev-cli-errors';
+import { run } from '@kbn/dev-cli-runner';
+import { ToolingLog } from '@kbn/tooling-log';
+import { getTimeReporter } from '@kbn/ci-stats-reporter';
+import { ErrorReporter } from '../utils';
+import { I18nCheckTaskContext } from '../types';
+import {
+  checkConfigs,
+  mergeConfigs,
+  extractDefaultMessagesTask,
+  integrateTranslations,
+  validateTranslationFiles,
+} from '../tasks';
+
+const toolingLog = new ToolingLog({
+  level: 'info',
+  writeTo: process.stdout,
+});
+
+const runStartTime = Date.now();
+const reportTime = getTimeReporter(toolingLog, 'scripts/i18n_check');
+
+run(
+  async ({
+    flags: { source, target, 'include-config': includeConfig, 'dry-run': dryRun },
+    log,
+  }) => {
+    if (typeof source !== 'string' || typeof target !== 'string') {
+      throw createFailError(
+        `${chalk.white.bgRed(' I18N ERROR ')} --target and --source options should be specified.`
+      );
+    }
+
+    if (typeof includeConfig === 'boolean') {
+      throw createFailError(
+        `${chalk.white.bgRed(' I18N ERROR ')} --include-config require a value`
+      );
+    }
+
+    if (typeof dryRun !== 'boolean') {
+      throw createFailError(`${chalk.white.bgRed(' I18N ERROR ')} --dry-run can't have a value`);
+    }
```

Comment:
Some of these flag validations are being repeated across various files, could they be moved to a central file and imported? I have added #187794 to look into making the validation automated.